### PR TITLE
Update ResponseType

### LINKED ISSUE #725: Add other variant into ResponseType
Shouldn't we add an `Other(i32)` variant into the [`ResponseType`](http://gtk-rs.org/docs/gtk/enum.ResponseType.html) enum? Docs says that gtk response types are below 0 but that users can provide custom ones. Shouldn't it be wrapped as well?

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #725.

cc @sdroege @EPashkin

### CODE REVIEW

File: src/response_type.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            -3 => ResponseType::Accept,
+            -4 => ResponseType::DeleteEvent,
+            -5 => ResponseType::Ok,
+            -6 => ResponseType::Cancel,
+            -7 => ResponseType::Close,
+            -8 => ResponseType::Yes,
+            -9 => ResponseType::No,
+            -10 => ResponseType::Apply,
+            -11 => ResponseType::Help,
+            value if value >= 0 => ResponseType::Other(value as u16),
```
[the remark this answers is not in the document]
Comment (sdroege) [reply src/response_type.rs]: Yes

**Comment by sdroege** [comment]
Looks good except for the one comment @EPashkin wrote

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/response_type.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    Ok,
+    Cancel,
+    Close,
+    Yes,
+    No,
+    Apply,
+    Help,
+    Other(u16),
+    #[doc(hidden)]
+    Unknown(i32),
```
Comment (sdroege) [inline src/response_type.rs]: Why did it lose its two underscores in the beginning?

Comment (GuillaumeGomez) [reply src/response_type.rs]: Want me to put them back? At first I just replaced it with `Other` and then put it back.

Comment (sdroege) [reply src/response_type.rs]: I'd prefer that with underscores, yes

**Comment by GuillaumeGomez** [comment]
Updated `Unknown`.

**Comment by sdroege** [comment]
:+1:

**Pull request merged**